### PR TITLE
Display lot barcode

### LINKED ISSUE #6290: Add lot barcodes to Lots in Stock Registry export
We need the following abilities:

1. We need to add the ability to download/print the barcode from the lot registry.
2. We need the ability to scan a barcode and find it in the Lot/Asset registry.
3. We need a column in the XLSX export of the lot registry called "barcode" that contains the barcode text so we can turn them into barcodes using external programs.

### PR DESCRIPTION
**Pull request by mbayopanda**
This PR adds : 
- A way for displaying and print individual lot barcode
image

image

- The barcode column in the lot registry
image

Close #6290

### CODE REVIEW

File: server/controllers/stock/reports/stock/lot_barcode/lot_barcode.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  _, ReportManager, db, identifiers, barcode, LOT_BARCODE_TEMPLATE,
+} = require('../../common');
+
+/**
+ * @method lotBarcodeReceipt
+ *
+ * @description
+ * This method displays the lot barcode
+ *
+ * GET /receipts/stock/lot_barcode/:uuid
```
Comment (jniles) [inline server/controllers/stock/reports/stock/lot_barcode/lot_barcode.js]: This is nice!  I would make this URL `/receipts/stock/lots/:uuid/barcode` just to be more rest-like :)

File: server/controllers/stock/core.js [hunk 2]
```diff
@@ -288,9 +290,11 @@ async function getLotsDepot(depotUuid, params, finalClause) {
 
   const resultFromProcess = await db.exec(query, queryParameters);
 
-  // add minumum delay
+  // add minumum delay and barcode
+  const { key } = identifiers.LOT;
```
Comment (jniles) [inline server/controllers/stock/core.js]: Nice 👍

File: client/src/modules/stock/lots/templates/action.cell.html [hunk 3]
```diff
@@ -12,7 +12,11 @@
         <i class="fa fa-edit"></i> <span translate>FORM.LABELS.EDIT</span>
       </a>
     </li>
-
+    <li>
```
Comment (jniles) [inline client/src/modules/stock/lots/templates/action.cell.html:16]: Can you put `<li class="separators">`s here to logically group the different parts of this menu together?

Something like:
  1. Everything that is a receipt goes together
  2. Everything that is a link goes together.
  3. Everything that causes an action (edit), goes together.
  4. If it is destructive (delete, cancel) it should be red and in its own section.

File: server/controllers/stock/reports/stock/lot_barcode/lot_barcode.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ */
+function lotBarcodeReceipt(req, res, next) {
+  let report;
+  const data = {};
+  const uuid = db.bid(req.params.uuid);
+  const options = {
+    filename : 'LOTS.BARCODE_FOR_LOT',
+    pageSize : 'A6',
```
Comment (jniles) [inline server/controllers/stock/reports/stock/lot_barcode/lot_barcode.js:19]: Does A7 exist?  :smile:

File: client/src/modules/stock/lots/registry.service.js [hunk 5]
```diff
@@ -46,6 +46,11 @@ function LotsRegistryService(uiGridConstants, Session) {
       displayName : 'TABLE.COLUMNS.INVENTORY_GROUP',
       headerTooltip : 'TABLE.COLUMNS.INVENTORY_GROUP',
       headerCellFilter : 'translate',
+    }, {
+      field : 'barcode',
+      displayName : 'LOTS.BARCODE',
+      headerTooltip : 'LOTS.BARCODE',
+      headerCellFilter : 'translate',
```
Comment (jniles) [inline client/src/modules/stock/lots/registry.service.js:53]: Can you put this in a cell template with a `bh-reference-link` that opens the barcode modal automatically when we click it?

Comment (jniles) [reply client/src/modules/stock/lots/registry.service.js:53]: Also, let's set `visible: false` by default.

**Review by jniles — CHANGES_REQUESTED** [review]
I made some basic comments, but the most important thing is that the barcode renderer doesn't seem to work on my computer!   Please investigate this.

firefox_rae64ltoQM

**Comment by jniles** [comment]
Nevermind, the barcode not rendering was an enterprise setting.  Thanks @mbayopanda for pointing this out!

File: server/controllers/stock/reports/stock/lot_barcode/lot_barcode.js (hunk 4 above)
Comment (mbayopanda) [reply server/controllers/stock/reports/stock/lot_barcode/lot_barcode.js:19]: No, it doesn't

**Comment by jniles** [comment]
@mbayopanda when you are comfortable that you are done with this, go ahead and merge it.

**Pull request merged**